examples/QIR/Optimization/README.md

```diff
@@ -0,0 +1,274 @@
+* (Linux only) Ensure the [GNU OpenMP support library](https://gcc.gnu.org/onlinedocs/libgomp/) is installed on your system, e.g. via `sudo apt install libgomp1`
+
+### Installing Clang
+
+Depending on your platform, some LLVM tools may only be available by compiling from source.
+*Clang* however should be readily available from package managers or as pre-compiled binaries on most systems.
+LLVM's [official release page](https://releases.llvm.org/download.html) provides sources and binaries of LLVM and Clang.
+It's recommended to use LLVM version 11.1.0, as this is the version used by the Q# compiler. 
+
+Here are some suggestions on how to obtain Clang for different platforms:
+
+* **Ubuntu** : `sudo apt install clang-11` (NOTE: this installs the command `clang-11` not `clang`)
+* **Windows** : `choco install llvm --version=11.1.0` using [chocolatey](https://chocolatey.org/)
+* **macOS** : `brew install llvm@11` using [homebrew](https://brew.sh/)
+
+Pre-built binaries/installers:
+
+* **Ubuntu** : get `clang+llvm-11.1.0-x86_64-linux-gnu-ubuntu-20.10.tar.xz` from the [GitHub release](https://github.com/llvm/llvm-project/releases/tag/llvmorg-11.1.0)
+* **Windows** : get `LLVM-11.1.0-win64.exe` from the [GitHub release](https://github.com/llvm/llvm-project/releases/tag/llvmorg-11.1.0)
+* **macOS** : get `clang+llvm-11.0.0-x86_64-apple-darwin.tar.xz` from the [11.0.0 release](https://github.com/llvm/llvm-project/releases/tag/llvmorg-11.0.0) (11.1.0 not released)
+
+### Installing the LLVM optimizer
+
+If you followed the instructions above for Linux or macOS, your LLVM installation will have included the *LLVM optimizer*.
+You can test this by typing `opt` or `opt-11` in your terminal.
+
+Unfortunately, `opt` is not part of the official Windows distribution, which means you will to need to [compile LLVM from source](https://llvm.org/docs/GettingStarted.html) to get it.
+You can still follow most of the guide without it though.
+
+## Generating QIR
+
+The starting point for QIR generation is a Q# program or library.
+This section details the process for a simple *hello world* program, but any valid Q# code can be substituted in its stead.
+
+### Creating a Q# project
+
+The first step is to create a .NET project for a standalone Q# application.
+The QDK project templates facilitate this task, which can be invoked with:
+
+```shell
+dotnet new console -lang Q# -o hello
+```
+
+Parameters: 
+
+* `console` : specify the project to be a standalone console application
+* `-lang Q#` : load the templates for Q# projects
+* `-o hello` : the project name, all files will be generated inside a folder of this name
+
+Other configurations are also possible, such as [Q# libraries with a C# host program](https://docs.microsoft.com/en-us/azure/quantum/install-csharp-qdk?tabs=tabid-cmdline%2Ctabid-csharp#creating-a-q-library-and-a-net-host).
+
+The standard Q# template produces a hello world program in the file `Program.qs`:
+
+```csharp
+namespace hello {
```

Comment:
This namespace doesn't follow the Q# style guide, as it starts with a lowercase letter.

Reply:
Fixed.